doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+
+#### Lifecycle
+
+```txt
+┌─────────────────┐ ┌───────────┐┌─────────────────┐     ┃
+│  Editor Plugin  │ │   Core    ││  Wizard Plugin  │     ┃  LEGEND
+└─────────────────┘ └───────────┘└─────────────────┘     ┃  ┌───────────┐
+         │                │               │              ┃  │  <name>   │   module with lifeline
+         │                │               │              ┃  └─────┬─────┘
+         │                │◀───register───│              ┃        │
+         │───register────▶│               │              ┃        │
+         │                │               │              ┃        │
+         │   requests a   │               │              ┃        │
+         │───wizard for──▶│               │              ┃
+         │    a given     │  asks wizard  │              ┃
+         │                │───if it can ─▶│              ┃  ──<desc.>  ─▶    Initiated action
+         │                │ handle given  │              ┃
+         │                │               │              ┃
+         │                │◀ ─ ─answer ─ ─│              ┃  ─ <answer>  ▶    Answer to an action
+         │                │               │              ┃
+         │                │   initiates   │              ┃
+         │                │    wizard     │              ┃            ┌─┐
+         │                │───with the ──▶│              ┃            │ │
+         │                │     given    ┌┴┐             ┃            │ │
+         │                │    element   │ │             ┃            │ │    Internal process
+         │                │              │ │wizarding    ┃            │ │
+         │                │              │ │process      ┃            │ │
+         │                │              │ │             ┃            └─┘
+         │                │              │ │             ┃
+         │                │              └┬┘             ┃
+         │                │               ├──┐           ┃
+         │                │               │  │           ┃
+         │                │               closes         ┃
+         │                │               itself         ┃
+         │                │               │  │           ┃
+         │                │               ◀──┘           ┃
+         │                │               │              ┃
+         X                X               X              ┃
+```
+
+##### Registration
+
+Registering a wizard plugin happens the same way as registering another plugin.
+We will add the wizard type to our `PluginSet`.
+```diff
+- export type PluginSet = { menu: Plugin[]; editor: Plugin[] };
++ export type PluginSet = { menu: Plugin[]; editor: Plugin[], wizard: Plugin[] };
+```
+
+Wizards must have two public static functions called `CanInspect` and `CanCreate`.
+`Core` will call these functions when it tries to figure out which wizard supports
+the given element and action.
+```ts
+type CanInspect: (tagName: string) => boolean
+type CanCreate: (tagName: string) => boolean
```

Comment:
Why not name these "Edit" and "Create", respectively? Also, isn't a name that starts with a lowercase letter more usual for methods?

```suggestion
Wizards must have two static methods called `canEdit` and `canCreate`.
`Core` will call these methods when it tries to figure out which wizard supports
the given element and action.
```ts
type canEdit: (tagName: string) => boolean
type canCreate: (tagName: string) => boolean
```

Reply:
I didn't see the "type" keyword in those lines. In general I wouldn't define named types for those, but wherever we do define named types, let's definitely capitalise them, since that's the TS and JS convention. Currently in open-scd, we don't capitalise any static fields, the same way open-wc, lit, and mwc do: They have all properties and methods start with lowercase regardless of whether they're defined on the class or on the instance, just like all constants and functions.

Reply:
I have tried to make the description clearer and already changed them to lowercase.
The types were just intended as a way of description, and I would only include it in the code if they help to make things more understandable.